src/mscorlib/src/System/Enum.cs

```diff
@@ -415,47 +418,53 @@ private static bool TryParseEnum(Type enumType, String value, bool ignoreCase, r
                 }
             }
 
-            String[] values = value.Split(enumSeperatorCharArray);
-
-            // Find the field.Lets assume that these are always static classes because the class is
-            //  an enum.
+            // Find the field. Let's assume that these are always static classes 
+            // because the class is an enum.
             ValuesAndNames entry = GetCachedValuesAndNames(rtType, true);
-
             String[] enumNames = entry.Names;
             ulong[] enumValues = entry.Values;
-            
-            for (int i = 0; i < values.Length; i++)
+
+            StringComparison comparison = ignoreCase ? 
+                StringComparison.OrdinalIgnoreCase : 
+                StringComparison.Ordinal;
+
+            for (int valueIndex = 0; valueIndex <= value.Length; valueIndex++)
```

Comment:
The equals is to handle the case where the string ends with a comma.

I can change it to a while instead of a for.